integration_tests/src/main/python/spark_init_internal.py

```diff
@@ -12,18 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from pyspark.sql import SparkSession
+try:
+    from pyspark.sql import SparkSession
+except ImportError as error:
+    import findspark
+    findspark.init()
+    from pyspark.sql import SparkSession
 
 def _spark__init():
     #Force the RapidsPlugin to be enabled, so it blows up if the classpath is not set properly
     # DO NOT SET ANY OTHER CONFIGS HERE!!!
     # due to bugs in pyspark/pytest it looks like any configs set here
     # can be reset in the middle of a test if specific operations are done (some types of cast etc)
     # enableHiveSupport() is needed for parquet bucket tests
-    _s = SparkSession.builder \
+    #TODO need to figure out a better way to do this optionally
+    import os
+    ALLOW_FRACTION = 1 / (int(os.getenv('RUN_PARALLEL')) + 2)
+    MAX_FRACTION = 1 / (int(os.getenv('RUN_PARALLEL')) + 1)
```

Comment:
So this does not appear to be going in the direction I would like simply because most of the hard coded configs here are not going to work on a regular cluster.  To be 100% clear I want one of two things. 

Option 1: Have a naming convention for environment variables that that this code can use to set spark configs.

i.e. `RAP_TEST_[config]=[value]`. 
```
spark_configs_map = {key[9:]: value for key, value in os.environ.items() if key.startswith('RAP_TEST_')}
```
Option 2: Have `run_pyspark_from_build.sh`  output a `spark-defaults.conf` file in a temporary directory with the configs in it that we care about for the maven build and then set `SPARK_CONF_DIR` to point to it. I don't like this one as much because we need to be careful about the temp directory and cleaning up, etc.

Either way we pick we also need to update the documentation to explain what is happening.

Reply:
@revans2 Updated the PR :
Move spark local mode conf of pytest parallelism into confMap

Add maven parameter '-Dpytest.RAP_TEST_PARALLEL=xx' to override the parallelism

 -Dpytest.RAP_TEST_PARALLEL=1 by default, it means not run xdist parallelism.

To run parallelism, set pytest.RAP_TEST_PARALLEL large than 1

In spark-rapids premerge pipeline, set 4 parallelism.